x-pack/legacy/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+import { sendEmail } from './lib/send_email';
+import { ActionType, ActionTypeExecutorOptions } from '../types';
+
+const PORT_MAX = 256 * 256 - 1;
+
+export type ActionTypeConfigType = TypeOf<typeof ActionTypeConfig.schema>;
+
+const ActionTypeConfig = {
+  path: 'actionTypeConfig',
+  schema: schema.object({
+    service: schema.maybe(schema.string()),
+    host: schema.maybe(schema.string()),
+    port: schema.maybe(schema.number({ min: 1, max: PORT_MAX })),
+    secure: schema.maybe(schema.boolean()),
+    user: schema.string(),
+    password: schema.string(),
+    from: schema.string(),
+  }),
+};
+
+export type ActionParamsType = TypeOf<typeof ActionParams.schema>;
+
+const ActionParams = {
+  path: 'actionParams',
+  schema: schema.object({
+    to: schema.arrayOf(schema.string(), { defaultValue: [] }),
+    cc: schema.arrayOf(schema.string(), { defaultValue: [] }),
+    bcc: schema.arrayOf(schema.string(), { defaultValue: [] }),
+    subject: schema.string(),
+    message: schema.string(),
+  }),
+};
+
+function validateConfig(object: any): { error?: Error; value?: ActionTypeConfigType } {
+  let value;
+
+  try {
+    value = ActionTypeConfig.schema.validate(object);
+  } catch (error) {
+    return { error };
+  }
+
+  const { service, host, port } = value;
+
+  // Make sure service is set, or if not, both host/port must be set.
+  // If service is set, host/port are ignored, when the email is sent.
+  if (service == null) {
+    if (host == null && port == null) {
+      return toErrorObject('either [service] or [host]/[port] is required');
+    }
+
+    if (host == null) {
+      return toErrorObject('[host] is required if [service] is not provided');
+    }
+
+    if (port == null) {
+      return toErrorObject('[port] is required if [service] is not provided');
+    }
+  }
+
+  return { value };
+}
+
+function validateParams(object: any): { error?: Error; value?: ActionParamsType } {
+  let value;
+
+  try {
+    value = ActionParams.schema.validate(object);
+  } catch (error) {
+    return { error };
+  }
+
+  const { to, cc, bcc } = value;
+  const addrs = to.length + cc.length + bcc.length;
+
+  if (addrs === 0) {
+    return toErrorObject('no [to], [cc], or [bcc] entries');
+  }
+
+  return { value };
+}
+
+export const actionType: ActionType = {
+  id: '.email',
+  name: 'email',
+  validate: {
+    config: { validate: validateConfig },
+    params: { validate: validateParams },
+  },
+  executor,
+};
```

Comment:
need to add `unencryptedAttributes` to the ActionType.  @mikecote I wonder if we should make this a required property vs optional, a little more in your face, so people will actually think about it, maybe not forget to enumerate them.

https://github.com/elastic/kibana/blob/005cf6f95449819eab81b6a098ab5f150744e737/x-pack/legacy/plugins/actions/server/types.ts#L40-L49

Reply:
I'm a fan of explicit over implicit, feel free to remove the `?` and such.